### PR TITLE
Merge upstream 5.7.20 into Percona Server

### PR DESCRIPTION
**Pull request by percona-ysorokin**

### CODE REVIEW

**Comment by percona-ysorokin** [comment]
Jenkins build link
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1312/

**Comment by percona-ysorokin** [comment]
The fix for
`keyring_vault.table_encrypt_4` and `rocksdb.index_merge_rocksdb2` (big)
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1317/

File: mysql-test/r/clone_consistent_snapshot.result [hunk 1]
```diff
@@ -26,11 +26,15 @@ ERROR HY000: Unknown thread id: 0
 # Functional tests
 COMMIT;
 SET SESSION tx_isolation='READ-COMMITTED';
+Warnings:
```
Comment (laurynas-biveinis) [inline mysql-test/r/clone_consistent_snapshot.result]: The testcase should be adjusted to transaction_isolation instead

File: mysql-test/r/percona_statement_set.result [hunk 2]
```diff
@@ -870,10 +870,16 @@ SET SESSION lock_wait_timeout = @old_lock_wait_timeout;
 Test for bug 1387951: SET STATEMENT ... FOR <statement> crashes server
 if <statement> is RW in a RO transaction
 SET @old_tx_read_only = @@session.tx_read_only;
+Warnings:
```
Comment (laurynas-biveinis) [inline mysql-test/r/percona_statement_set.result]: The testcase should be edited to transaction_read_only instead (and no need to save/restore its session value, just reset to false in the end)

File: mysql-test/suite/rocksdb/r/delete.result [hunk 3]
```diff
@@ -1,4 +1,6 @@
 set session tx_isolation='READ-COMMITTED';
+Warnings:
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/r/delete.result]: The testcase should be adjusted to transaction_isolation

File: mysql-test/suite/rocksdb/r/delete_ignore.result [hunk 4]
```diff
@@ -1,4 +1,6 @@
 set session tx_isolation='READ-COMMITTED';
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/r/delete_ignore.result]: Likewise

File: mysql-test/suite/rocksdb/r/delete_quick.result [hunk 5]
```diff
@@ -1,4 +1,6 @@
 set session tx_isolation='READ-COMMITTED';
+Warnings:
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/r/delete_quick.result]: Likewise

File: mysql-test/suite/rocksdb/r/insert.result [hunk 6]
```diff
@@ -1,4 +1,6 @@
 set session tx_isolation='READ-COMMITTED';
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/r/insert.result]: Likewise

File: mysql-test/suite/rocksdb/r/issue100.result [hunk 7]
```diff
@@ -1,4 +1,6 @@
 set session tx_isolation='READ-COMMITTED';
+Warnings:
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/r/issue100.result]: Likewise

File: mysql-test/suite/rocksdb/r/issue100_delete.result [hunk 8]
```diff
@@ -1,4 +1,6 @@
 set session tx_isolation='READ-COMMITTED';
+Warnings:
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/r/issue100_delete.result]: Likewise for all the RocksDB tests that set tx_isolation themselves and not through a common include file

File: scripts/mysqld_safe.sh [hunk 9]
```diff
@@ -215,10 +215,6 @@ eval_log_error () {
 
   #echo "Running mysqld: [$cmd]"
   eval "$cmd"
-  ret=$?
```
Comment (laurynas-biveinis) [inline scripts/mysqld_safe.sh:218]: What's the status of BLD-807 after this revert?

File: sql/sql_plugin.cc [hunk 10]
```diff
@@ -2253,6 +2253,19 @@ static bool mysql_uninstall_plugin(THD *thd, const LEX_STRING *name)
              "Stop any active semisynchronous I/O threads on this slave first.");
     goto err;
   }
+
+  /* If Group Replication is in use, the plugin can't be uninstalled.
+   * The command STOP GROUP_REPLICATION should be used before uninstall.
+  */
+  if (plugin->ref_count && !strcmp(name->str, "group_replication"))
```
Comment (laurynas-biveinis) [inline sql/sql_plugin.cc:2260]: seriously

(no this is not a review comment, I just couldn't move along silently)

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Testcases updates requested for easier 8.0 port.

A Jenkins rerun only for affected tests is required (probably easiest to do at suite granularity)

File: sql/sql_plugin.cc (hunk 10 above)
Comment (laurynas-biveinis) [reply sql/sql_plugin.cc:2260]: https://bugs.mysql.com/bug.php?id=88597

File: scripts/mysqld_safe.sh (hunk 9 above)
Comment (percona-ysorokin) [reply scripts/mysqld_safe.sh:218]: Created a new BLD-893 and marked as a release blocker

**Comment by percona-ysorokin** [comment]
Jenkins re-run (`--suite=rocksdb --big-test`)
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1322
Jenkins re-run (`percona_statement_set clone_consistent_snapshot rpl_rocksdb_snapshot --repeat=2`)
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1324/

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**